file/writable_file_writer.h

```diff
@@ -195,7 +195,8 @@ class WritableFileWriter {
     TEST_SYNC_POINT_CALLBACK("WritableFileWriter::WritableFileWriter:0",
                              reinterpret_cast<void*>(max_buffer_size_));
     buf_.Alignment(writable_file_->GetRequiredBufferAlignment());
-    buf_.AllocateNewBuffer(std::min((size_t)65536, max_buffer_size_));
+    // Removed to reduce WAL writer memory usage.
+    // buf_.AllocateNewBuffer(std::min((size_t)65536, max_buffer_size_));
```

Comment:
Does it affect raftstore v1？

Reply:
Maybe we can keep the code and use a smaller `max_buffer_size` in raftstore v2.

Reply:
The allocation is delayed to actual write, which I think is good for both v1 and v2. Because the initialization is also on critical path, moving it elsewhere amortizes the overhead.

Reply:
I see. How about skipping alloc is `max_buffer_size_ == 0`?

```suggestion
    if max_buffer_size_ != 0 {
      buf_.AllocateNewBuffer(std::min((size_t)65536, max_buffer_size_));
    }
```

Reply:
/cc https://github.com/tikv/tikv/issues/15275

Reply:
Oh, I forgot this flag works both for WAL and SST. So we can't simply change the config.